benchmarks/definition/base.py

```diff
@@ -46,48 +44,45 @@ class BenchmarkSettings(ABC, BenchmarkSerialization):
     random_seed: int = field(validator=instance_of(int), default=1337)
     """The used random seed."""
 
-    runmode: str | None = field(
-        validator=optional(instance_of(str)),
-        default=None,
-        converter=lambda x: RunMode.STANDARD.value if x is None else x,
+    runmode: RunMode = field(
+        validator=instance_of(RunMode),
+        default=RunMode.STANDARD,
     )
     """Mode to run benchmark in.
     Can be used to adjust the selection of setting parameters.
     """
 
+    def set_runmode(self, runmode: RunMode) -> None:
+        """Set the runmode for the benchmark settings."""
+        if runmode not in RunMode:
```

Comment:
My plan was to use RunMode as "Necesary" run modes, while users may add more. Since I also use it to check that all runmodes are present when benchmark is created. So maybe we can change RunMode to MandatoryRunModes and allow users to just add more as desired? - And these may then be added only for some benchmarks. Because now the validator for RunMode based params would err out if all benchmarks dont implement all run modes, even if runing with a benchmark_list that only needs some benchmarks